go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1600,7 +1600,7 @@ func convertErrorCode(err error) vtrpcpb.Code {
 		sqlerror.ERTooLongString, sqlerror.ERDelayedInsertTableLocked, sqlerror.ERDupUnique, sqlerror.ERRequiresPrimaryKey, sqlerror.ERCantDoThisDuringAnTransaction, sqlerror.ERReadOnlyTransaction,
 		sqlerror.ERCannotAddForeign, sqlerror.ERNoReferencedRow, sqlerror.ERRowIsReferenced, sqlerror.ERCantUpdateWithReadLock, sqlerror.ERNoDefault, sqlerror.EROperandColumns,
 		sqlerror.ERSubqueryNo1Row, sqlerror.ERNonUpdateableTable, sqlerror.ERFeatureDisabled, sqlerror.ERDuplicatedValueInType, sqlerror.ERRowIsReferenced2,
-		sqlerror.ErNoReferencedRow2, sqlerror.ERWarnDataOutOfRange:
+		sqlerror.ErNoReferencedRow2, sqlerror.ERWarnDataOutOfRange, sqlerror.ERInnodbIndexCorrupt:
 		errCode = vtrpcpb.Code_FAILED_PRECONDITION
 	case sqlerror.EROptionPreventsStatement:
```

Comment:
This got added under Internal Error in `constants` but used as `Failed_Precondition`.
```
// internal
	ERInternalError = ErrorCode(1815)
	ERInternalError      = ErrorCode(1815)
	ERInnodbIndexCorrupt = ErrorCode(1817)
```

We should align them.